integration-tests/emulated/tests/governance/polkadot/src/lib.rs

```diff
@@ -14,18 +14,99 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use emulated_integration_tests_common::impls::{assert_expected_events, bx, TestExt};
-use frame_support::assert_ok;
-use integration_tests_helpers::Chain;
+use emulated_integration_tests_common::{
+	impls::{assert_expected_events, bx, Encode, TestExt},
+	xcm_emulator::Chain,
+};
+use frame_support::{
+	assert_ok,
+	dispatch::{DispatchResultWithPostInfo, PostDispatchInfo},
+};
 use polkadot_system_emulated_network::CollectivesPolkadotPara as CollectivesPolkadot;
-use sp_runtime::traits::Dispatchable;
+use sp_core::H256;
+use sp_runtime::traits::{Dispatchable, Hash};
 use xcm::{latest::prelude::*, VersionedLocation, VersionedXcm};
 
 #[cfg(test)]
 mod open_gov_on_asset_hub;
 #[cfg(test)]
 mod open_gov_on_relay;
 
+/// Wraps a runtime call in a whitelist preimage call and dispatches it
+pub fn dispatch_whitelisted_call_with_preimage<T>(
+	call: T::RuntimeCall,
+	origin: T::RuntimeOrigin,
+) -> DispatchResultWithPostInfo
+where
+	T: Chain,
+	T::Runtime: pallet_whitelist::Config,
+	T::RuntimeCall: From<pallet_whitelist::Call<T::Runtime>>
+		+ Into<<T::Runtime as pallet_whitelist::Config>::RuntimeCall>
+		+ Dispatchable<RuntimeOrigin = T::RuntimeOrigin, PostInfo = PostDispatchInfo>,
+{
+	T::execute_with(|| {
+		let whitelist_call: T::RuntimeCall =
+			pallet_whitelist::Call::<T::Runtime>::dispatch_whitelisted_call_with_preimage {
+				call: Box::new(call.into()),
+			}
+			.into();
+		whitelist_call.dispatch(origin)
+	})
+}
+
+/// Encodes a runtime call, stores its preimage (dispatch), and returns its H256 hash
+pub fn dispatch_store_preimage_call<T>(call: T::RuntimeCall) -> H256
```

Comment:
@karolk91 wow, very nice, I think `dispatch_whitelisted_call_with_preimage` and `dispatch_store_preimage_call` we can move directly to the `emulated-integration-tests-common` and reused for polkadot-sdk and fellows

Just one very nit - a you can merge
```suggestion
/// Encodes a runtime call, stores it as a preimage, and returns its H256 hash
pub fn dispatch_note_preimage_call<T>(call: T::RuntimeCall) -> H256
```